src/annotator/anchoring/range.js

```diff
@@ -0,0 +1,490 @@
+      normalRange.commonAncestor = normalRange.commonAncestor.parentNode;
+    }
+
+    return new Range.NormalizedRange(normalRange);
+  }
+
+  /**
+   * Creates a range suitable for storage.
+   *
+   * root           - A root Element from which to anchor the serialization.
+   * ignoreSelector - A selector String of elements to ignore. For example
+   *                  elements injected by the annotator.
+   *
+   * Returns an instance of SerializedRange.
+   */
+  serialize(root, ignoreSelector) {
+    return this.normalize().serialize(root, ignoreSelector);
+  }
+};
+
+/**
+ * A normalized range is most commonly used throughout the annotator.
+ * its the result of a deserialized SerializedRange or a BrowserRange with
+ * out browser inconsistencies.
+ */
+Range.NormalizedRange = class NormalizedRange {
+  /**
+   * Creates an instance of a NormalizedRange.
+   *
+   * This is usually created by calling the .normalize() method on one of the
+   * other Range classes rather than manually.
+   *
+   * obj - An Object literal. Should have the following properties.
+   *       commonAncestor: A Element that encompasses both the start and end nodes
+   *       start:          The first TextNode in the range.
+   *       end             The last TextNode in the range.
+   *
+   * Returns an instance of NormalizedRange.
+   */
+  constructor(obj) {
+    this.commonAncestor = obj.commonAncestor;
+    this.start = obj.start;
+    this.end = obj.end;
+  }
+
+  /**
+   * For API consistency.
+   *
+   * Returns itself.
+   */
+  normalize() {
+    return this;
+  }
+
+  /**
+   * Limits the nodes within the NormalizedRange to those contained
+   * withing the bounds parameter. It returns an updated range with all
+   * properties updated. NOTE: Method returns null if all nodes fall outside
+   * of the bounds.
+   *
+   * bounds - An Element to limit the range to.
+   *
+   * Returns updated self or null.
+   */
+  limit(bounds) {
+    const nodes = $.grep(
+      this.textNodes(),
+      node => node.parentNode === bounds || $.contains(bounds, node.parentNode)
+    );
+    if (!nodes.length) {
+      return null;
+    }
+
+    this.start = nodes[0];
+    this.end = nodes[nodes.length - 1];
+
+    const startParents = $(this.start).parents();
+
+    for (let parent of $(this.end).parents()) {
+      if (startParents.index(parent) !== -1) {
+        this.commonAncestor = parent;
+        break;
+      }
+    }
+    return this;
+  }
+
+  /**
+   * Convert this range into an object consisting of two pairs of (xpath,
+   * character offset), which can be easily stored in a database.
+   *
+   * root -           The root Element relative to which XPaths should be calculated
+   * ignoreSelector - A selector String of elements to ignore. For example
+   *                  elements injected by the annotator.
+   *
+   * Returns an instance of SerializedRange.
+   */
+  serialize(root, ignoreSelector) {
+    const serialization = function (node, isEnd) {
```

Comment:
```suggestion
    const serialization = (node, isEnd) => {
```

This function doesn't reference `this`, so this can be an arrow function.